encoding/protoavro/unmarshal.go

```diff
@@ -35,8 +35,25 @@ func (m *Unmarshaler) Unmarshal(message proto.Message) error {
 	if err != nil {
 		return fmt.Errorf("read message: %w", err)
 	}
-	if err := decodeJSON(data, message); err != nil {
+	if err := decodeJSON(data, message, &UnmarshalOptions{}); err != nil {
 		return fmt.Errorf("decode message: %w", err)
 	}
 	return nil
 }
+
+type UnmarshalOptions struct {
+	MarshalOptions *MarshalOptions
+	TrashCan []proto.Message
+}
```

Comment:
Instead of including `MarshalOptions` as field we could have a `ConvertToUnmarshalOptions` method for `MarshalOptions`.